cpu/atmega_common/periph/cpuid.c

```diff
@@ -44,24 +37,12 @@
  */
 void cpuid_get(void *id)
 {
-    uint8_t  signature_0 = boot_signature_byte_get(0x00);
-    uint8_t  signature_1 = boot_signature_byte_get(0x02);
-    uint8_t  signature_2 = boot_signature_byte_get(0x04);
-
-    uint8_t  usr_sign_0 = boot_signature_byte_get(0x0100);
-    uint8_t  usr_sign_1 = boot_signature_byte_get(0x0102);
-
-    uint8_t addr[CPUID_LEN] = {
-            signature_0,        /* 0x1E Atmel manufacturer ID */
-            signature_1,        /* 0xA8 Part Number high byte */
-            signature_2,        /* 0x02 Part Number low byte */
-            MAN_ID_0,           /* 0x1F Atmel JEDEC manufacturer ID */
-            PART_NUM,           /* 0x94 PART_NUM  Identification */
-            VERSION_NUM,        /* 0x02 VERSION_NUM  Identification */
+    uint8_t *out = id;
+    out[0] = boot_signature_byte_get(0x00);     /* Atmel manufacturer ID (0x1E) */
+    out[1] = boot_signature_byte_get(0x02);     /* Part Number high byte */
+    out[2] = boot_signature_byte_get(0x04);     /* Part Number low byte */
+    out[3] = boot_signature_byte_get(0x01);     /* internal RC calibration byte */
 /* last two bytes can be set to flash page 1. for differentiation of different boards */
-            usr_sign_0,         /* user signature 0 */
-            usr_sign_1,         /* user signature 1 */
-    };
-
-    memcpy(id, addr, CPUID_LEN);
+    out[4] = boot_signature_byte_get(0x0100);   /* user signature 0 */
+    out[5] = boot_signature_byte_get(0x0102);   /* user signature 1 */
```

Comment:
Searching the data sheets of the ATmega32U4, ATmega328P, ATmega1281, ATmega1284P and the ATmega2560 for the string "user signature" yields no matches. It seems they are only present on the ATmega128RFA1 and the ATmega256RFR2. (Or the feature as a different name for the older MCUs.)

Reply:
Well I can drop them then.
Addresses `0x0`, `0x1`, `0x2` and `0x4` are specified for all Atmel parts though.

I would argue that the unspecified in-between values won't change either - for IDing individual parts they might be good enough. 

On ATmega328P `0x2` can even have two meanings - wut?
![download](https://user-images.githubusercontent.com/1301112/68088290-e510ae00-fe5d-11e9-92a7-a7cb341e8328.png)

Reply:
Now it relies only on addresses available on all ATmega MCUs.